internal/datacoord/segment_manager.go

```diff
@@ -189,25 +198,44 @@ func newSegmentManager(meta *meta, allocator allocator, opts ...allocOption) *Se
 	return manager
 }
 
+// loadSegmentsFromMeta generate corresponding segment status for each segment from meta
 func (s *SegmentManager) loadSegmentsFromMeta() {
 	segments := s.meta.GetUnFlushedSegments()
 	ids := make([]UniqueID, 0, len(segments))
 	for _, seg := range segments {
 		ids = append(ids, seg.ID)
 		stat := &segmentStatus{
-			id:             seg.ID,
-			collectionID:   seg.CollectionID,
-			partitionID:    seg.PartitionID,
-			total:          seg.MaxRowNum,
-			allocations:    []*allocation{},
-			insertChannel:  seg.InsertChannel,
-			lastExpireTime: seg.LastExpireTime,
-			sealed:         seg.State == commonpb.SegmentState_Sealed,
+			id:          seg.ID,
+			allocations: make([]*allocation, 0, 16),
 		}
 		s.stats[seg.ID] = stat
 	}
 	log.Debug("Restore segment allocation", zap.Int64s("segments", ids))
 }
+
+// getAllocation unified way to retrieve allocation struct
+func (s *SegmentManager) getAllocation(numOfRows int64, expireTs uint64) *allocation {
+	v := s.allocPool.Get()
+	if v == nil {
+		return &allocation{
+			numOfRows:  numOfRows,
+			expireTime: expireTs,
+		}
+	}
+	a, ok := v.(*allocation)
+	if !ok {
+		a = &allocation{}
+	}
```

Comment:
Would it be possible that allocPool return nil if we have a New function? Or we can just write`s.allocPool.Get().(*allocation)`. I am not sure.